### ISSUE TITLE
1.20.4 Login event never triggered on Node 18

### ISSUE THREAD

**Issue by Opisek** [PrismarineJS/node-minecraft-protocol#1289]
## Versions
 - minecraft-protocol: 1.47.0
 - node: 18.19.1

## Detailed description of a problem
The protocol is broken on Node 18. Works fine on Node 20. The simplest way to reproduce it is to simply listen for a `login` event.

## Current code
```js
import { createServer } from "minecraft-protocol";

const options = {
   host: "127.0.0.1",
   motd: "test",
   port: 25565,
   maxPlayers: 20,
   version: "1.20.4"
}

const srv = createServer(options);

srv.on("login", client => {
   console.log("join");
});
```

## Expected behavior
"join" is printed on the console when a client tries to join the server.

## Actual behavior
Nothing is printed and one of the following happens
- Client sees "Disconnected"
- Client sees "IndexOutOfBounds"
- Client sees "codecCorruptedFrameException"

Furthermore, the server may receive the following error (does not always happen either):
```
Disconnecting client because error Error: read ECONNRESET at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -104,
  code: 'ECONNRESET',
  syscall: 'read'
}
```

## Additional context
I'd like to ask you to adjust the supported engine value for this library to disclude older versions of Node, if this issue can't be fixed easily.

**Comment by rom1504** [comment]
Please enable debug mode and post packet log

**Comment by rom1504** [comment]
minecraft-protocol { reason: '{"text":"DidNotEncryptVerifyTokenProperly"}' } +0ms

this is clearly the problem

**Comment by rom1504** [comment]
@Opisek can you reproduce in a vanilla server?

**Comment by Opisek** [comment]
No I cannot. The issue is only with node-minecraft-protocol based server, I can join any other server just fine. The issue persisted on two machines I tried for a server and two different minecraft users from different countries.

**Comment by rom1504** [comment]
Are you using online login?

**Comment by Opisek** [comment]
May I know what Node version you use for testing?

Here's how the node-minecraft-protocol behaves on my systems depending on the Node version:
v18.19.1 (npm v10.2.4): Doesn't work
v20.10.0 (npm v10.2.3): Works
v20.11.1 (npm v10.2.4): Doesn't work

(Tested on both a Debian and a Windows server with both the example code in my original post and the linked production code.)

**Comment by rom1504** [comment]
https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/.github/workflows/ci.yml#L20 we use node 18

However we disabled the online login tests a while back because it did not work well with the throttling limits on Mojang auth https://github.com/PrismarineJS/node-minecraft-protocol/issues/372

Might be time to revisit this...

**Comment by Hyphrio** [comment]
Weirdly enough on my testing, Deno 1.42.0 (on node compat) emits the Login event correctly.

**Comment by Khaogamermain01** [comment]
I was trying to make a proxy and suddenly could not join any Minecraft-Protocol Server so I tested multiple versions of Minecraft such as `1.12.2`, `1.16.5`, and `1.19.2` and none of them worked.

When I join the server from the client it says:
- Logging In
- Encrypting
- Disconnected

After I found this GitHub issue I tried v20.10.0 and it worked.

Node v18.0.0: Yes
Node v18.20.2: No
Node v19.0.0: Yes
Node v19.9.0: No
Node v20.0.0: Yes
Node v20.10.0: Yes
Node v20.12.2: No
Node v21.0.0: Yes
Node v21.7.3: No (Error)
`DeprecationWarning: The 'punycode' module is deprecated.`
```
Internal Exception: io.netty.handler.codec.DecoderException:
java.lang.IndexOutOfBoundsException: Index: 94, Size: 5
```
Node v22.0.0: No

**Comment by Khaogamermain01** [comment]
Any update on this?

**Comment by ytcalifax** [comment]
Is the library still being actually updated? This seems to have been an issue for months with no commits trying to fix it.

**Comment by ytcalifax** [comment]
I am (i have been postponing doing the project I have in mind for quite a while already), it’s just starting to look like this library is either not being updated anymore or there are not enough trusted maintainers that can keep up with the bugs (this bug is now several months old and is a breaking one) or even merge the pull requests.

**Comment by qwqtoday** [comment]
> I am (i have been postponing doing the project I have in mind for quite a while already), it’s just starting to look like this library is either not being updated anymore or there are not enough trusted maintainers that can keep up with the bugs (this bug is now several months old and is a breaking one) or even merge the pull requests.

I had an idea of forking all prismarinejs projects to another GitHub organization to get my wanted changes to be merged. Most of the prismarinejs projects doesn't really need to have to update once a while, usually they are just classes and types for the core projects. Because of this, they didn't really have to add many maintainers for most of the projects. But I agree with you that they didn't keep up with the bugs recently.

**Comment by extremeheat** [comment]
> Is the library still being actually updated? This seems to have been an issue for months with no commits trying to fix it.

Indeed, this is normal for open source projects; you don't just push commits for the sake of doing commits to appear active. That doesn't mean there are no maintainers, it just means there are no PRs to merge.

If you (or someone) make a PR to fix it then there is a discussion about not merging it, then we can talk about why the PR isn't merged, but at this time there is no PR to fix it.

There are indeed no maintainers here being paid to monitor and fix your bugs, and we are not sponsored by Microsoft or some other org. The commits done here are done by volunteers and will be done as soon as someone has time and is interested in doing a PR. You are always free to fork, but that obviously won't fix bugs unless you write the code to fix them yourself.

> Any update on this?

Could be related to #1293

**Comment by Opisek** [comment]
What is the state of this? Has the root cause been determined? The whole package appears to be unusable beyond a certain minecraft and node.js version.

**Comment by soshimee** [comment]
https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/src/server/login.js#L128
`TypeError: RSA_PKCS1_PADDING is no longer supported for private decryption`

**Comment by extremeheat** [comment]
Should be fixed with https://github.com/PrismarineJS/node-minecraft-protocol/pull/1319

**Comment by ArmoredFuzzball** [comment]
This remains unfixed, now with Node v22.13, nmp 1.54.0, and Minecraft 1.21.X.
```
import { createServer } from "minecraft-protocol";

const options = {
    host: "127.0.0.1",
    motd: "test",
    port: 25565,
    maxPlayers: 20,
    version: "1.21.3"
}

const server = createServer(options);

server.on("login", client => {
    console.log("join");
});
```
No printing of join, and Minecraft client fails with: Failed

**Comment by extremeheat** [comment]
The server implementation is working correctly, and flying-squid supports 1.21 by using nmp on latest node v22. If you have a problem you need to figure out how flying-squid implements the server and do the same in your program, or use flying-squid.

The apparent node version issue here causing the `DidNotEncryptVerifyTokenProperly` for OP should have been fixed with #1319
